### DISCUSSION TITLE
Joomla Big Spinner / Preloader Image

### DISCUSSION THREAD

**Discussion by webchun** [joomla/joomla-cms#47160] [2. Cold Features]
### Steps to reproduce the issue
I think the default joomla spinner / preloader image size is too big and the feel is so 2010-ish / old school. I think it'll be better to change it with other one that has a modern look or maybe simply use a standard one, so it will give a netral impression, especially for first user/adopter, because this spinner also appear in the installation process.![screen shot 2020-07-14 at 15 21 49](https://issues.joomla.org/uploads/1/d79fe6df428bf6fe4e923f5c92085611.jpg)

**Comment by richard67** [comment]
> I think the default joomla spinner / preloader image size is too big and the feel is so 2010-ish / old school. I think it'll be better to > change it with other one that has a modern look or maybe simply use a standard one, so it will give a netral impression, especially > for first user/adopter, because this spinner also appear in the installation process.

I think that's just your personal taste. Other people might like the big spinner with Joomla logo.

**Comment by PhilETaylor** [comment]
My bug bear here is that (most) people are not going to that page for "install from web" and so having to wait 5seconds for the loader to finish, just to change tab to one of the more common tabs is a painful wait... 

Install from web should be demoted to the non active tabs or promoted to its own place as a "flagship feature" and its own menu item/link.

**Comment by infograf768** [comment]
First thing I do when creating a J site is change the order of web installer. 😺

**Comment by brianteeman** [comment]
Pull requests welcome - will only take a few minutes to make that change

**Comment by astridx** [comment]
You most likely opened this issue to hear if others agree, right?
The spinner is not the point with the highest priority for me at the moment. But I don't have to have it either.
Maybe you have a better suggestion. Then a PR would be great and I would test it.

**Comment by chmst** [comment]
my2cent: It is very easy to change the order of tabs with an PR. 
But not a real solution - which user group do we prefer? Developers or power users who like to hide or move the "install-from-web" plugin (a few clicks per installation). Or newbie users who download Joomla and make their own website - and so can see at once which extensions they can have?

I'd like to set this on the list for a "easy Joomla" project we have already spoken about. For power user - deactivate the plugin, for normal users let it be in the foreground.

**Comment by PhilETaylor** [comment]
Im not saying remove it or reorder it, but the "loading" screen doesn't need to cover the whole page and impede navigation, when it relates only to the "install from web" tab only. It could be limited to that tab and thus not get in the way of navigation

At the moment it impedes navigation, and takes as long as it wants based on the speed of a remote service loading. 

Sometimes that is 5+ seconds. 

Even with my 1Gbps Fibre connection it takes over 3 seconds in every one of my last 5 page load tests. 

For "users" on "normal" connections (remember mobile first?) this could be a lot longer. 

The loading of `https://appscdn.joomla.org/webapps/index.php?format=json&option=com_apps&view=dashboard&product=Sm9vbWxhIQ==&release=NC4w&dev_level=MC1iZXRhMg==&list=grid&lang=ZW4tR0I=` consistently takes 1-2seconds - then depending on the browser, injecting all that html into the page takes time too.

**Comment by webchun** [comment]
> I think that's just your personal taste. Other people might like the big spinner with Joomla logo.

That's exactly my point, other people might be like and other might be not. 

> My bug bear here is that (most) people are not going to that page for "install from web" and so having to wait 5seconds for the loader to finish, just to change tab to one of the more common tabs is a painful wait...

Another one that is bugging me is when you open the extension's detail page from a list view (category view), you can't go back to the previous opened list view by pressing browser's back button.

Back to my issue, I'm talking something like these, a smaller one 
![0](https://user-images.githubusercontent.com/4277547/87613324-36847300-c737-11ea-8b2b-0dba8883316a.jpg)

Anyway, I created some designs ideas. Please let me know what you're thinking

BW, with block area behind the logo to add more contrast and focus to the logo
![image](https://user-images.githubusercontent.com/4277547/87612741-95e18380-c735-11ea-8010-1a211e12277d.png)

Same as before, colored version 
![2](https://user-images.githubusercontent.com/4277547/87612982-389a0200-c736-11ea-9518-cc78f8354f60.jpg)

A minimal look
![4](https://user-images.githubusercontent.com/4277547/87612979-3637a800-c736-11ea-8aa3-981744c4de18.jpg)

A minimal look, colored version
![3](https://user-images.githubusercontent.com/4277547/87612980-3768d500-c736-11ea-80ff-b05424d8454d.jpg)

A simplest one, dots with joomla colors
![5](https://user-images.githubusercontent.com/4277547/87612977-35067b00-c736-11ea-9f74-148537f948b8.jpg)

Same as before with block area
![6](https://user-images.githubusercontent.com/4277547/87612971-30da5d80-c736-11ea-906b-606af591f4d5.jpg)

Same as before, alternate dots position 
![7](https://user-images.githubusercontent.com/4277547/87613048-5cf5de80-c736-11ea-8c0f-2766beeefe25.jpg)

**Comment by C-Lodder** [comment]
> The loading of https://appscdn.joomla.org/webapps/index.php?format=json&option=com_apps&view=dashboard&product=Sm9vbWxhIQ==&release=NC4w&dev_level=MC1iZXRhMg==&list=grid&lang=ZW4tR0I= consistently takes 1-2seconds - then depending on the browser, injecting all that html into the page takes time to

I was wondering why on earth the API returns markup. Without it, the response is **60x** smaller

**Comment by mbabker** [comment]
Because the original system designers were adamant on building a system where the intermediate server would cache the JED API data, render the markup with caching support on the server, and deliver the rendered markup via CDN.  Instead of building a proper API system.  That would’ve been too easy.

And, well, I can’t say I’m bothered enough to spend my time rewriting a component that myself and Peter van Westen are the only ones to have made significant effort to improve over the last 3 years into a more properly designed architecture that I would still somehow be the only person to invest any time supporting.

**Comment by ghost** [comment]
Where this Joomla Big Spinner / Preloader Image is? I want to test your version

**Comment by 4digitalworld** [comment]
> 
> 
> ![image](https://user-images.githubusercontent.com/4277547/87612741-95e18380-c735-11ea-8010-1a211e12277d.png)
> 
> ![2](https://user-images.githubusercontent.com/4277547/87612982-389a0200-c736-11ea-9518-cc78f8354f60.jpg)

I think these 2 examples are very good, they show a loading indicator, which is very important, especially if there are problems with the internet.

**Comment by ghost** [comment]
Where this Joomla code this Big Spinner / Preloader Image is located ? I want to test it

**Comment by PhilETaylor** [comment]
Its used on the Install from Web tab in Joomla Extension Installer, and also in the Joomla installation process.

**Comment by dgrammatiko** [comment]
The spinner is now SVG based and it supports dark/light mode. Also in this particular case the size could also be adjust.
This Issue probably is resolved

![Image](https://github.com/user-attachments/assets/9ff06b24-e573-4014-908c-c2f584d45b78)
